### DISCUSSION TITLE
Cache: Purge cache manually

### DISCUSSION THREAD

**Discussion by sand97** [vercel/next.js#32975] [Help] [answered]
### Problem
If you export an async function called **getStaticProps** from a page, Next.js will pre-render this page at build time using the props returned by **getStaticProps**. If you would like to update the cache, next provide **revalidate** property that is an optional amount in seconds after which a page re-generation can occur. That is not so good because:

1. The getStaticProps function is called unnecessarily and the same page is regenerate and store in the cache.
2. When the data(content of blog for example) that depend page change, the page don’t change directly and some users can continue to read outdated information.
3. It is sometimes difficult to choose a number of seconds for re-validation. If you make a blog for example, some articles will need to be updated faster than others.

### Proposition
Next.js can provide boolean(default false) property **purge** in getStaticProps payload. 
When some URL like https://www.website.com/article/slug-1 is call with DELETE method and purge property is true in payload of getStaticProps in articles/[slugs].js, Next.js delete the cache. 

If routes is called with PUT method, and purge is true Next.js delete the cache and regenerate the page.  

### Alternative
I have write an article that explain how we can purge the cache of Next.js specific route. 
https://medium.com/@briceguekam/next-js-cache-optimization-7f14d2113f74#8995-5d7d7acfede9

**Comment by leerob** [comment, marked as answer]
I'm incredibly excited to share that Next.js 12.1 was just released, which includes **on-demand Incremental Static Regeneration**! We know many of you have been looking forward to this for a while, including myself, and I'm so happy it is here. 

- [Learn about Next.js 12.1](https://nextjs.org/blog/next-12-1)
- [Watch a demo of on-demand ISR](https://youtu.be/BGexHR1tuOA)
- [View another demo and give us your feedback](https://on-demand-isr.vercel.app/)
- [Read the documentation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)
